flink-core/src/main/java/org/apache/flink/api/common/typeinfo/BasicArrayTypeInfo.java

```diff
@@ -121,6 +122,18 @@ public boolean isKeyType() {
 	}
 
 	@Override
+	@PublicEvolving
+	public <F> FieldAccessor<T, F> getFieldAccessor(int pos, ExecutionConfig config) {
+		return new FieldAccessor.ArrayFieldAccessor<T, F>(pos, this);
+	}
+
+	@Override
+	@PublicEvolving
+	public <F> FieldAccessor<T, F> getFieldAccessor(String field, ExecutionConfig config) {
+		throw new InvalidFieldReferenceException("Selecting a field from an array type is only supported by index.");
```

Comment:
We could also add an additional constructor in `ArrayFieldAccessor` that takes a string and parses the int position there. Then we don't need to throw exceptions. What do you think?